integrationTests/vm/arwen/utils.go

```diff
@@ -137,15 +149,63 @@ func (context *TestContext) DeploySC(wasmPath string, parametersString string) {
 
 	err := context.TxProcessor.ProcessTransaction(tx)
 	if err != nil {
-		require.FailNow(context.T, err.Error())
+		return err
+	}
+
+	err = context.GetSilentSCProcessorError()
+	if err != nil {
+		return err
+	}
+
+	owner.Nonce++
+
+	_, err = context.Accounts.Commit()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UpgradeSC -
+func (context *TestContext) UpgradeSC(wasmPath string, parametersString string) error {
+	scCode := GetSCCode(wasmPath)
+	owner := &context.Owner
+
+	codeMetadataHex := hex.EncodeToString(context.ScCodeMetadata.ToBytes())
+	txData := strings.Join([]string{"upgradeContract", scCode, codeMetadataHex}, "@")
+	if parametersString != "" {
+		txData = txData + "@" + parametersString
+	}
+
+	tx := &transaction.Transaction{
+		Nonce:    owner.Nonce,
+		Value:    big.NewInt(0),
+		RcvAddr:  context.ScAddress,
+		SndAddr:  owner.Address,
+		GasPrice: 1,
+		GasLimit: math.MaxInt32,
+		Data:     []byte(txData),
+	}
+
+	err := context.TxProcessor.ProcessTransaction(tx)
+	if err != nil {
+		return err
+	}
+
+	err = context.GetSilentSCProcessorError()
+	if err != nil {
+		return err
 	}
 
 	owner.Nonce++
 
```

Comment:
delete empty line. save the owner account nonce with context.Accounts.SaveAccount(owner) ?

Reply:
Removed empty line - for the `owner` (`testParticipant`) see comment above.